From 393aa3554ad4c8322794f28d09ede0d3814c4401 Mon Sep 17 00:00:00 2001
From: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
Date: Tue, 14 Mar 2017 17:11:13 -0400
Subject: [PATCH 1969/2094] Revert "drm/amdgpu: Use AMDGPU_FENCE_OWNER_VM in
 unmap"

This reverts commit 6bb00e491bfb ("drm/amdgpu: Use AMDGPU_FENCE_OWNER_VM
in unmap")

Change-Id: Ie525e713269ddec8dedcdb5d4f0f2e9dcedffcbe
Signed-off-by: Harish Kasiviswanathan <Harish.Kasiviswanathan@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
index e92ae53..1a43253 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_vm.c
@@ -1358,6 +1358,10 @@ static int amdgpu_vm_bo_update_mapping(struct amdgpu_device *adev,
 
 	ring = container_of(vm->entity.sched, struct amdgpu_ring, sched);
 
+	/* sync to everything on unmapping */
+	if (!(flags & AMDGPU_PTE_VALID))
+		owner = AMDGPU_FENCE_OWNER_UNDEFINED;
+
 	nptes = last - start + 1;
 
 	/*
-- 
2.7.4

